src/resolution/composite-arns-resolver.ts

```diff
@@ -74,61 +79,68 @@ export class CompositeArNSResolver implements NameResolver {
       };
     }
 
-    if (arnsBaseNamesCache.has(baseName)) {
-      try {
-        // check if our resolution cache contains the FULL name
-        const cachedResolutionBuffer = await this.cache.get(name);
-        if (cachedResolutionBuffer) {
-          const cachedResolution: NameResolution = JSON.parse(
-            cachedResolutionBuffer.toString(),
-          );
-          resolution = cachedResolution; // hold on to this in case we need it
-          // use the override ttl if it exists, otherwise use the cached resolution ttl
-          const ttlSeconds = this.overrides?.ttlSeconds ?? cachedResolution.ttl;
-          if (
-            cachedResolution !== undefined &&
-            cachedResolution.resolvedAt !== undefined &&
-            ttlSeconds !== undefined &&
-            cachedResolution.resolvedAt + ttlSeconds * 1000 > Date.now()
-          ) {
-            metrics.arnsCacheHitCounter.inc();
-            this.log.info('Cache hit for arns name', { name });
-            return cachedResolution;
-          }
+    try {
+      // check if our base name is in our arns names cache, this triggers a debounce with a ttl dependent on if it's in the cache or not
+      const baseNameInCache =
+        await this.arnsNamesCache.getCachedArNSBaseName(baseName);
+
+      if (!baseNameInCache) {
+        throw new Error('Base name not found in ArNS names cache');
```

Comment:
i will modify this to return the empty resolution data right away instead of throwing an error and a warning log

Reply:
[91cdc3d](https://github.com/ar-io/ar-io-node/pull/264/commits/91cdc3d186784a01c94f857b0ec0ec204369fa22)